### DISCUSSION TITLE
[firebase_messaging] iOS Direct Channel API will no longer be supported (used by firebase_messaging)

### DISCUSSION THREAD

**Discussion by red42** [firebase/flutterfire#5608] [Feature request]
**Describe the bug**
I received an email from Firebase that says the iOS Direct Channel API will no longer be supported, but the firebase_messaging plugin seems to use that. Is there a planned migration for the plugin?

Here's the content of the email:

> We are writing to let you know that Firebase Messaging iOS Direct Channel API support is marked for deprecation. Firebase Messaging iOS SDK version 7.0 will be a breaking change where the SDK will no longer support iOS Direct Channel API.
> 
> What do I need to know?
> Firebase Messaging iOS Direct Channel has been used for sending data to apps when they are in the foreground and for upstream messages. The Firebase Messaging iOS SDK v7.0 (slated to be released in September 2020) will no longer support the iOS Direct Channel API. Your app will break if you update to the Firebase Messaging iOS SDK v7.0 without migrating off the iOS Direct Channel API.
> 
> We are removing this feature in favor of natively supported messaging features provided by Apple Push Notification Service.
> 
> To prevent disruptions to your users, we recommend that you discontinue use of the iOS Direct Channel API as soon as possible. Please note that for the time being, your existing users can still continue to receive messages via the iOS Direct Channel API. However, within the next year, we will follow up this announcement with plans to deprecate the backend service support which may break your app with older versions of the Messaging iOS SDK (prior to v7.0).

The relevant code in `FLTFirebaseMessagingPlugin.m` is:
```
  } else if ([@"configure" isEqualToString:method]) {
    [FIRMessaging messaging].shouldEstablishDirectChannel = true;
    [[UIApplication sharedApplication] registerForRemoteNotifications];
    if (_launchNotification != nil && _launchNotification[kGCMMessageIDKey]) {
      [_channel invokeMethod:@"onLaunch" arguments:_launchNotification];
    }
...
```

The changelog for the plugin states:
## 5.0.1+1

* Enable support for `onMessage` on iOS using `shouldEstablishDirectChannel`.

So this seems to be actively used.

**Flutter doctor**
Run `flutter doctor` and paste the output below:

```
[✓] Flutter (Channel stable, v1.17.1, on Mac OS X 10.15.4 19E287, locale en-BR)
    • Flutter version 1.17.1 at /Users/red42/flutter
    • Framework revision f7a6a7906b (10 days ago), 2020-05-12 18:39:00 -0700
    • Engine revision 6bc433c6b6
    • Dart version 2.8.2

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/red42/Library/Android/sdk
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = /Users/red42/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.1

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.2

[✓] Connected device (1 available)
    • My iPhone • 1163974e4a6e4af3506e39396c8fabc9e8ca7b79 • ios • iOS 12.4.5

• No issues found!
Process finished with exit code 0
```

**Comment by QkLahpita** [comment]
Are there any solutions for this issue?

**Comment by Alissonpcl** [comment]
Any solutions for this? I have a production app that uses this plugins to receive push messages and a started receiving the same e-mail from Google.

**Comment by ih-cedricf** [comment]
Same concern here. I have a production app that uses this plugin.

**Comment by sarope** [comment]
Any news on a solution?
